proto/spire/api/server/local_authority/v1/local_authority.proto

```diff
@@ -0,0 +1,174 @@
+
+        // RevokeX509Authority revokes the previously active X.509 authority by
+        // removing it from the bundle and propagating this update throughout
+        // the cluster.
+        //
+        // If a previously active X.509 authority does not exist (e.g. if one
+        // has been prepared but not activated yet), a FailedPrecondition
+        // error will be returned.
+        rpc RevokeX509Authority(RevokeX509AuthorityRequest) returns (RevokeX509AuthorityResponse);
+}
+
+message GetJWTAuthorityStateRequest {}
+
+message GetJWTAuthorityStateResponse {
+        repeated AuthorityState states = 1;
+}
+
+message PrepareJWTAuthorityRequest {}
+
+message PrepareJWTAuthorityResponse {
+        AuthorityState prepared_authority = 1;
+}
+
+message ActivateJWTAuthorityRequest {}
+
+message ActivateJWTAuthorityResponse {
+        AuthorityState activated_authority = 1;
+}
+
+message TaintJWTAuthorityRequest {}
+
+message TaintJWTAuthorityResponse {
+        AuthorityState tainted_authority = 1;
+}
+
+message RevokeJWTAuthorityRequest {}
+
+message RevokeJWTAuthorityResponse {
+        AuthorityState revoked_authority = 1;
+}
+
+message GetX509AuthorityStateRequest {}
+
+message GetX509AuthorityStateResponse {
+        repeated AuthorityState states = 1;
+}
+
+message PrepareX509AuthorityRequest {}
+
+message PrepareX509AuthorityResponse {
+        AuthorityState prepared_authority = 1;
+}
+
+message ActivateX509AuthorityRequest {}
+
+message ActivateX509AuthorityResponse {
+        AuthorityState activated_authority = 1;
+}
+
+message TaintX509AuthorityRequest {}
+
+message TaintX509AuthorityResponse {
+        AuthorityState tainted_authority = 1;
+}
+
+message RevokeX509AuthorityRequest {}
+
+message RevokeX509AuthorityResponse {
+        AuthorityState revoked_authority = 1;
+}
+
+message AuthorityState {
+        // An opaque name for the "slot" that this authority is housed in
+        string slot = 1;
+
+        // The authority's public key, PKIX encoded
+        bytes public_key = 2;
+
+        enum Status {
+                // Key status is unknown. This is an error condition and should
+                // never really be encountered.
+                UNKNOWN = 0;
+
+                // This slot is empty. To fill it, prepare a new authority.
+                EMPTY = 1;
+
+                // This is a new authority that has been added to the bundle
+                // but is not yet in use.
+                PREPARED = 2;
+
+                // This authority is active and is currently being used for
+                // signing operations.
+                ACTIVE = 3;
+
+                // This is an old authority that was previously in use but is
+                // no longer.
+                OLD = 4;
+
+                // This is an old authority that was previously in use but is
+                // now expired.
+                EXPIRED = 5;
```

Comment:
not sure we  want to expose expired authorities, since we'll automatically remove them